azure-pipelines-dependency-graph-testing.yml

```diff
@@ -239,6 +268,39 @@ stages:
               summaryFileLocation: '$(System.DefaultWorkingDirectory)/**/coverage.xml'
               reportDirectory: '$(System.DefaultWorkingDirectory)/**/htmlcov'
 
+  - stage: usage_stats_integration
+    dependsOn: [scope_check, lint]
+    pool:
+      vmImage: 'ubuntu-latest'
+
+    jobs:
+      - job: test_usage_stats_messages
+        condition: eq(stageDependencies.scope_check.changes.outputs['CheckChanges.GEChanged'], true)
+        variables:
+          python.version: '3.8'
+
+        steps:
+          - task: UsePythonVersion@0
+            inputs:
+              versionSpec: '$(python.version)'
+            displayName: 'Use Python $(python.version)'
+
+          - bash: python -m pip install --upgrade pip==20.2.4
+            displayName: 'Update pip'
+
+          - script: |
+              pip install --requirement requirements-dev-base.txt
+              pip install --requirement requirements.txt
+              pip install .
+            displayName: 'Install dependencies'
+
+          # Due to the relatively small number of usage_stats tests, we deem it appropriate to test them in their entirely through pytest
```

Comment:
Usage stats are already pretty complicated and aren't a bottleneck so let's leave them alone.